webapp/src/components/issue_attribute_selector.jsx

```diff
@@ -0,0 +1,150 @@
+        load: PropTypes.func,
+        theme: PropTypes.object.isRequired,
+        value: PropTypes.oneOfType([
+            PropTypes.object,
+            PropTypes.array,
+            PropTypes.string,
+        ]),
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            options: [],
+            values: [],
+            isLoading: false,
+            invalid: false,
+            error: '',
+        };
+    }
+
+    componentDidMount() {
+        if (this.props.addValidate) {
+            this.props.addValidate(this.isValid);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.props.removeValidate) {
+            this.props.removeValidate(this.isValid);
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.invalid && this.props.value !== prevProps.value) {
+            this.setState({invalid: false}); //eslint-disable-line react/no-did-update-set-state
+        }
+
+        if (prevProps.repo !== this.props.repo) {
+            this.loadOptions();
+        }
+    }
+
+    loadOptions = async () => {
+        this.setState({isLoading: true});
+        const options = await this.props.load();
+
+        // filter out currently selected options that do not exist in the new repo
+        const optionValues = options.map((option) => option.value);
+        const validValues = this.state.values.filter((value) => optionValues.includes(value.value));
+
+        this.setState({
+            options,
+            values: validValues,
+            isLoading: false,
+        });
+    }
+
+    onChange = (values) => {
+        this.setState({values});
+
+        if (!values) {
+            this.props.onChange(this.props.isMulti ? [] : '');
+            return;
+        }
+
+        this.props.onChange(this.props.isMulti ? values.map((v) => v.value) : values.value);
+
+        if (this.props.resetInvalidOnChange) {
+            this.setState({invalid: false});
+        }
+    };
+
+    isValid = () => {
+        if (!this.props.required) {
+            return true;
+        }
+
+        const valid = Boolean(this.props.value && this.props.value.toString().length !== 0);
+        this.setState({invalid: !valid});
+        return valid;
+    };
+
+    errorComponent = () => {
+        if (!this.state.error) {
+            return null;
+        }
+
+        return (
+            <p className='alert alert-danger'>
+                <i
+                    className='fa fa-warning'
+                    title='Warning Icon'
+                />
+                <span> {this.state.error.toString()}</span>
+            </p>
+        );
+    };
+
+    validationError = () => {
+        if (!this.props.required || !this.state.invalid) {
+            return null;
+        }
+
+        return (
+            <p className='help-text error-text'>
+                <span>{'This field is required.'}</span>
+            </p>
+        );
+    };
+
+    render = () => {
```

Comment:
Not important, but we usually declare the `render` function is this way

```suggestion
    render() {
```